mmv1/third_party/terraform/provider/provider.go.tmpl

```diff
@@ -416,6 +416,13 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 		return nil, diag.FromErr(err)
 	}
 
+	// Bypass checkDestroyer failure with universe domain set
```

Comment:
@c2thorn FYI, this is added to bypass check destroy error

Reply:
Why does this work exactly? I don't think we're generally expected to call `d.Set` inside the provider as it's not a full resource, so I'm a bit surprised it even works.

Reply:
If this works, this is the exact same as removing the check later in https://github.com/hashicorp/terraform-provider-google/blob/main/google/provider/provider.go#L1266. There's no point in having both.

I was hoping to either set the universe domain in a test-only setup function, or skip the check altogether by adding something to 
```
else if config.UniverseDomain != "" && config.UniverseDomain != "googleapis.com"
```
that verifies if it is a test run.

Reply:
What is the issue we're seeing exactly? Does https://github.com/hao-nan-li/magic-modules/blob/2b5b82e44b188362545a5ca43a854c7397079d75/mmv1/third_party/terraform/provider/universe/universe_domain_compute_test.go#L145 not have the domain set?

Reply:
Here's the flow: 
1) https://github.com/hashicorp/terraform-provider-google/blob/main/google/provider/universe/universe_domain_compute_test.go#L100
2) https://github.com/hashicorp/terraform-provider-google/blob/main/google/acctest/provider_test_utils.go#L45
3) The configure function then calls https://github.com/hashicorp/terraform-provider-google/blob/ff829fe23a4235e6566fd07be74c52e32e376b69/google/provider/provider.go#L921

Finally an error occurs because universe_domain is not set in d (schema.ResourceData). And the code right below will return an error, then the check destroyer function fails.

Adding this block of code works around this problem

Reply:
Discussed offline; we can likely set the value on https://github.com/hashicorp/terraform-provider-google/blob/8ef7e5fbcf4f8d5a2a141dfff29bffc52d119e65/google/acctest/provider_test_utils.go#L44 to later successfully call configure.

More broadly, we should figure out how the two initialization paths (through the provider factory and there) are different, and ideally align them.

Reply:
Applied fix in provider_test_utils.

Will take a look at the second suggestion.